### PR TITLE
Running time entry card animations

### LINKED ISSUE #322: Animations: Card - stopping a running entry
**Stopping a running time entry**
Stop button scales down and entry moves away with the card.

card-running-stop

```javascript
toggl.stop.onTap ->
	hideIcon(toggl.stop)
	hideEntry()
	hideCard()

hideIcon = (icon) ->
	icon.animate
		scale: 0
		options:
			time: LeaveTimingFaster
			curve: EaseIn

hideEntry = () ->
	toggl.entry.animate
		opacity: 0
		y: 568
		options:
			time: LeaveTiming
			curve: EaseIn

hideCard = () ->
	toggl.card.animate
		y: 568
		options:
			time: LeaveTiming
			curve: cardOutCurve
```

### LINKED ISSUE #326: Animations: Card - closing a running entry
**Closing a running time entry**
Keyboard and card move away, overlay and content fade away, close button scales down. After .1 delay stop button, arrow and entry content appear.

card-running-close

```javascript
toggl.close.onTap ->
	hideKeyboard()
	RemoveOverlay()
	moveCard(512)
	hideIcon(toggl.close)
	hideContent()
	Utils.delay .1, ->
		showIcon(toggl.stop)
		showEntry()

hideKeyboard = () ->
	toggl.Keyboards.animate
		y: 568
		options:
			time: LeaveTiming
			curve: EaseIn

RemoveOverlay = () ->
	toggl.overlay.animate
		opacity: 0
		options:
			time: LeaveTiming
			curve: EaseIn

moveCard = (y) ->
	toggl.card.animate
		y: y
		options:
			time: EnterTiming
			curve: cardInCurve

hideIcon = (icon) ->
	icon.animate
		scale: 0
		options:
			time: LeaveTimingFaster
			curve: EaseIn

hideContent = () ->
	toggl.content.animate
		opacity: 0
		options:
			time: LeaveTimingFaster
			curve: EaseIn

showIcon = (icon) ->
	icon.animate
		scale: 1
		options:
			time: EnterTiming
			curve: EaseOut

showEntry = () ->
	toggl.entry.animate
		opacity: 1
		y: 524
		options:
			time: EnterTiming
			curve: EaseOut
```

### LINKED ISSUE #990: Log view: add padding at the bottom
Currently when user scrolls at the bottom of the log list, the last time entry is hidden behind the running time entry (or play button). To avoid that happening, padding between the last TE and bottom of the screen should be 64pt.

Zeplin: https://zpl.io/a309jlQ

Current:
img_4593

Better solution:
screen shot 2017-11-20 at 09 51 48

### PR DESCRIPTION
**Pull request by willsb**
Closes #326 
Closes #322 
Closes #990 

Please do a full functional test. DO YOUR BEST TO BREAK THIS :dragon:

Squash whenever

### CODE REVIEW

**Comment by arturs105** [comment]
For issue #990: the space between the running entry card and last time entry looks too small. Also, shouldn't there be padding at the bottom, even when there is no running entry (so that start button would be on top of last entry)?

When continuing an entry, a gray bar appears where the running entry card will be. Can't really see in the gif due to colors, but try out on simulator with slow animations (⌘+T) and you'll see it. The same grey bar can be briefly seen, when stopping and entry and when closing the start time entry view (better seen with slow animations turned on).
dec-07-2017 10-37-16

**Comment by willsb** [comment]
The second point of your review is easily addressed by syncing both animations.

For the first point, though, I'd like @pe0ny to clarify what should happen when the Play button is visible, since the zeplin layouts linked in the issue only show the behavior for the stop button

**Comment by pe0ny** [comment]
@willsb Am I missing something or what do you mean by what should happen when the play button is visible? Do you mean how the running card appears is correct or...?

**Comment by pe0ny** [comment]
@willsb asked about if we need padding at the bottom when timer is not running, and the answer is yes we do. Otherwise, last time entry's continue button is not visible.

File: Toggl.Daneel/Extensions/AnimationExtensions.cs [hunk 1]
```diff
@@ -24,7 +24,18 @@ public static void Animate(double duration, CubicBezierCurve curve, Action chang
...
             if (completion != null)
                 buttonAnimator.AddCompletion(_ => completion());
-            
+
+            buttonAnimator.StartAnimation();
+        }
+
+        public static void Animate(double duration, nfloat delay, CubicBezierCurve curve, Action changes, Action completion = null)
+        {
+            var buttonAnimator = new UIViewPropertyAnimator(duration,  curve.ToCubicTimingParameters());
```
Comment (arturs105) [inline Toggl.Daneel/Extensions/AnimationExtensions.cs]: Why is this called `buttonAnimator`?

File: Toggl.Daneel/Extensions/AnimationExtensions.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ public static UICubicTimingParameters ToCubicTimingParameters(this CubicBezierCu
             => new UICubicTimingParameters(
                 new CGPoint(self.P0, self.P1),
                 new CGPoint(self.P2, self.P3)
-            );
+        );
 
         public static void Animate(double duration, CubicBezierCurve curve, Action changes, Action completion = null)
```
Comment (arturs105) [inline Toggl.Daneel/Extensions/AnimationExtensions.cs]: Can't you call the other `Animate` method with `delay: 0`?

File: Toggl.Daneel/ViewControllers/MainViewController.cs [hunk 3]
```diff
@@ -110,10 +101,24 @@ public override void ViewDidLoad()
...
+            if (!viewInitialized) return;
+
+            if (visible)
+            {
+                showTimeEntryCard();
+            }
+            else
+            {
+                hideTimeEntryCard();
+            }
```
Comment (arturs105) [inline Toggl.Daneel/ViewControllers/MainViewController.cs]: These parentheses can be removed. Or they can stay. Your call.

File: Toggl.Daneel/ViewControllers/MainViewController.cs [hunk 4]
```diff
@@ -193,11 +193,46 @@ private void prepareViews()
...
+                    );
+                });
+        }
+
         private void animateSpider()
         {
             SpiderBroImageView.Transform = CGAffineTransform.MakeRotation(-animationAngle);
 
-            UIView.Animate(Animation.Timings.SpiderBro, 0, UIViewAnimationOptions.Autoreverse | UIViewAnimationOptions.Repeat, 
+            UIView.Animate(SpiderBro, 0, UIViewAnimationOptions.Autoreverse | UIViewAnimationOptions.Repeat,
```
Comment (arturs105) [inline Toggl.Daneel/ViewControllers/MainViewController.cs]: I liked the previous name better.

File: Toggl.Daneel/Extensions/AnimationExtensions.cs (hunk 1 above)
Comment (willsb) [reply Toggl.Daneel/Extensions/AnimationExtensions.cs]: 👀

File: Toggl.Daneel/ViewControllers/MainViewController.cs (hunk 3 above)
Comment (willsb) [reply Toggl.Daneel/ViewControllers/MainViewController.cs]: I'd rather have them here

**Pull request merged**